lib/crypto.js

```diff
@@ -282,7 +282,16 @@ Sign.prototype.sign = function sign(options, encoding) {
 
   var key = options.key || options;
   var passphrase = options.passphrase || null;
-  var ret = this._handle.sign(toBuf(key), null, passphrase, options);
+
+  // Options specific to RSA
+  var rsaPadding = options.padding || constants.RSA_PKCS1_PADDING;
+  var pssSaltLength = constants.RSA_PSS_SALTLEN_MAX_SIGN;
+  if (typeof options.saltLength === 'number') {
+    pssSaltLength = options.saltLength;
+  }
```

Comment:
This should probably be a little stricter in what it accepts and throw a TypeError when the input is invalid.  For example, what happens when I pass `{ padding: -0.5 }`?